sdk/program/src/secp256k1_recover.rs

```diff
@@ -103,7 +103,10 @@ pub fn secp256k1_recover(
             .map_err(|_| Secp256k1RecoverError::InvalidHash)?;
         let recovery_id = libsecp256k1::RecoveryId::parse(recovery_id)
             .map_err(|_| Secp256k1RecoverError::InvalidRecoveryId)?;
-        let signature = libsecp256k1::Signature::parse_standard_slice(signature)
+
```

Comment:
This is not used internally and we probably want external users to get the fix right away.  Why not leave this as `parse_standard_slice`?

Reply:
My initial plan was to keep the client-side and program-side consistent,  let the client slip after feature activation, then be punctual about removing the gating and switching the client-side over

Reply:
As discussed, I went ahead and changed the (likely unused today) client-side non-BPF implementation to anticipate the feature activation.  This'll just drive us to activate faster :wink: